droidlet/lowlevel/hello_robot/remote/remote_hello_robot_ros.py

```diff
@@ -0,0 +1,271 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+"""
+# python -m Pyro4.naming -n <MYIP>
+import select
+import logging
+import os
+import json
+import time
+import copy
+from math import *
+import math
+from rich import print
+import Pyro4
+import numpy as np
+from droidlet.lowlevel.hello_robot.remote.utils import transform_global_to_base, goto
+from stretch_ros_move_api import MoveNode as Robot
+from droidlet.lowlevel.pyro_utils import safe_call
+import traceback
+
+
+Pyro4.config.SERIALIZER = "pickle"
+Pyro4.config.SERIALIZERS_ACCEPTED.add("pickle")
+Pyro4.config.ITER_STREAMING = True
+
+
+# #####################################################
+@Pyro4.expose
+class RemoteHelloRobot(object):
+    """Hello Robot interface"""
+
+    def __init__(self, ip):
+        self._ip = ip
+        self._robot = Robot()
+        self._robot.start()
+        self._done = True
+        self.cam = None
+        # Read battery maintenance guide https://docs.hello-robot.com/battery_maintenance_guide/
+        self._load_urdf()
+        self.tilt_correction = 0.0
+
+    def _load_urdf(self):
+        import os
+
+        urdf_path = os.path.join(
+            os.getenv("HELLO_FLEET_PATH"),
+            os.getenv("HELLO_FLEET_ID"),
+            "exported_urdf",
+            "stretch.urdf",
+        )
+
+        from pytransform3d.urdf import UrdfTransformManager
+        import pytransform3d.transformations as pt
+        import pytransform3d.visualizer as pv
+        import numpy as np
+
+        self.tm = UrdfTransformManager()
+        with open(urdf_path, "r") as f:
+            urdf = f.read()
+            self.tm.load_urdf(urdf)
+
+    def set_tilt_correction(self, angle):
+        """
+        angle in radians
+        """
+        print(
+            "[hello-robot] Setting tilt correction " "to angle: {} degrees".format(degrees(angle))
+        )
+
+        self.tilt_correction = angle
+
+    def get_camera_transform(self):
+        head_pan = self.get_pan()
+        head_tilt = self.get_tilt()
+
+        if self.tilt_correction != 0.0:
+            head_tilt += self.tilt_correction
+
+        # Get Camera transform
+        self.tm.set_joint("joint_head_pan", head_pan)
+        self.tm.set_joint("joint_head_tilt", head_tilt)
+        camera_transform = self.tm.get_transform("camera_color_frame", "base_link")
+
+        # correct for base_link's z offset from the ground
+        # at 0, the correction is -0.091491526943
+        # at 90, the correction is +0.11526719 + -0.091491526943
+        # linear interpolate the correction of 0.023775
+        interp_correction = 0.11526719 * abs(head_tilt) / radians(90)
+        # print('interp_correction', interp_correction)
+
+        camera_transform[2, 3] += -0.091491526943 + interp_correction
+
+        return camera_transform
+
+    def get_base_state(self):
```

Comment:
The `yaw` is in `[-3.14, 3.14]` when using Hector SLAM and in `[0, 6.28]` when not, maybe we could make it consistent?

Reply:
done. added a `math.pi` to the yaw in this case, so that it's consistent to `[0, 6.28]`.